support/cas-server-support-openrewrite/build.gradle

```diff
@@ -1,38 +1,37 @@
 description = "Apereo CAS Open Rewrite Support"
 
-processResources {
-    def casVersion = project.version
+
+tasks.register("createRecipe") {
+    def casVersion = rootProject.version
     def tomcatVersion = project.tomcatVersion
     def sourceCompatibility = project.sourceCompatibility
     def springBootVersion = project.springBootVersion
     def jibVersion = project.jibVersion
     def gradleVersion = gradle.gradleVersion
     def recipeName = "cas${casVersion.replaceAll(/\.|-SNAPSHOT/, '')}"
     def templateContents = file("src/main/resources/current.gtemplate").text
-
-    def outputDir = file("src/main/resources/META-INF/rewrite")
-    def outputFile = file("${outputDir.absolutePath}/${recipeName}.yml")
-    outputs.file(outputFile)
+    def recipeFileName = "$buildDir/generated/META-INF/rewrite/${recipeName}.yml"
 
     doFirst {
         def model = [:]
-
         model.put("recipeName", recipeName)
         model.put("tomcatVersion", tomcatVersion)
         model.put("casVersion", casVersion)
         model.put("javaVersion", sourceCompatibility)
         model.put("jibVersion", jibVersion)
         model.put("gradleVersion", gradleVersion)
         model.put("springBootVersion", springBootVersion)
-        
+
         def engine = new groovy.text.GStringTemplateEngine()
         def recipe = engine.createTemplate(templateContents).make(model).toString()
 
-        if (!outputDir.exists()) {
-            outputDir.mkdirs()
-        }
+        def outputFile = file(recipeFileName)
+        outputFile.parentFile.mkdirs()
         outputFile.write(recipe)
     }
 }
 
-resourcesJar.dependsOn(processResources)
+jar {
+    dependsOn(createRecipe)
+    from 'build/generated'
```

Comment:
Is there a specific reason why you are adding this only to the `jar` task, instead of adding it to the (main) sourceSet? You could do that with `sourceSets.main.output.dir file("$buildDir/generated"), builtBy: 'createRecipe'`.

Reply:
Thanks much! Made the change.